src/github_changelog/semver.clj

```diff
@@ -1,23 +1,137 @@
 (ns github-changelog.semver
   (:require [clj-semver.core :as semver]
-            [clojure.string :as str]))
+            [clojure.spec.alpha :as s]
+            [clojure.spec.gen.alpha :as gen]
+            [clojure.string :as str]
+            [github-changelog.spec :as spec]))
+
+(defn- map->str [m]
+  (cond-> (->> (map m [:major :minor :patch]) (str/join \.))
+    (:pre-release m) (str "-" (:pre-release m))
+    (:build m) (str "+" (:build m))))
```

Comment:
I would use destructuring in the arglist of `map->str` to make this a bit more DRY.